Add some more namedtensor builds to the CI

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #21648 Expose torch.empty(sizes, *, names, ...) to Python
* #21647 Implement at::empty(IntArrayRef, DimnameList?, TensorOptions) in aten
* #21636 Add some named tensor helper functions
* **#21632 Add some more namedtensor builds to the CI**
* #21633 Add virtual dtor to NamedTensorMetaInterface

Added:
- important ASAN build (runs on PRs, but only if the keyword is
specified).
- important cuda py2.7 build

Test Plan:
- [namedtensor ci]

gh-metadata: pytorch pytorch 21632 gh/zou3519/44/head

Differential Revision: [D15760331](https://our.internmc.facebook.com/intern/diff/D15760331)

zou3519 committed:
Add some more namedtensor builds to the CI

Added:
- important ASAN build (runs on PRs, but only if the keyword is
specified).
- unimportant cuda 2.7 build (runs on master).

Test Plan:
- [namedtensor ci]

zou3519 committed:
Update on "Add some more namedtensor builds to the CI"

Add some more namedtensor builds to the CI

Added:
- important ASAN build (runs on PRs, but only if the keyword is
specified).
- unimportant cuda 2.7 build (runs on master).

Test Plan:
- [namedtensor ci]

gh-metadata: pytorch pytorch 21632 gh/zou3519/44/head

zou3519 committed:
Update on "Add some more namedtensor builds to the CI"

Add some more namedtensor builds to the CI

Added:
- important ASAN build (runs on PRs, but only if the keyword is
specified).
- unimportant cuda 2.7 build (runs on master).

Test Plan:
- [namedtensor ci]

gh-metadata: pytorch pytorch 21632 gh/zou3519/44/head

zou3519 committed:
Update on "Add some more namedtensor builds to the CI"

Add some more namedtensor builds to the CI

Added:
- important ASAN build (runs on PRs, but only if the keyword is
specified).
- important cuda py2.7 build

Test Plan:
- [namedtensor ci]

gh-metadata: pytorch pytorch 21632 gh/zou3519/44/head

ezyang reviewed (approved):
I dunno if this actually works; let's see!

zou3519 committed:
Update on "Add some more namedtensor builds to the CI"

Add some more namedtensor builds to the CI

Added:
- important ASAN build (runs on PRs, but only if the keyword is
specified).
- important cuda py2.7 build

Test Plan:
- [namedtensor ci]

gh-metadata: pytorch pytorch 21632 gh/zou3519/44/head
